wgsl/index.bs

```diff
@@ -3753,18 +3753,18 @@ all properties of the members of *S*, including attributes, carry over to the me
   </xmp>
 </div>
 
-## Type Declaration Grammar ## {#type-declarations}
+## Type Specifier Grammar ## {#type-specifiers}
 
 <div class='syntax' noexport='true'>
-  <dfn for=syntax>type_decl</dfn> :
+  <dfn for=syntax>type</dfn> :
```

Comment:
The PR title says this is `type_specifier` why not use that and then we still have `type` as the token available?

Reply:
I forgot to edit the title- description is more clear, apologies!

Reply:
Well, my preference would be to call this `type_specifier` as was the original title, heh. That leaves `type` free as the token name.

Reply:
I'd explicitly favor type, akin to Rust and Go.

Rust: https://doc.rust-lang.org/reference/statements.html#let-statements

Go: https://go.dev/ref/spec#Variable_declarations

Reply:
I would prefer using "type_specifier" as the grammar element where an author writes the name of a type.
"type" is also a particular keyword for creating a type alias.  Let's not name a grammar element for something that is a keyword used for a different purpose.

Reply:
@dneto0 If I rename to type_specifier, can I still inline that one 'type' to the construction? I think that will set a helpful precedent for having some headroom to iterate upon the readability of both grammar-in-spec and products-of-grammar quickly. It will also expose a new parameter for us to be able to tweak further with alignment to modern languages and their reference/spec preferences.

Reply:
Chaning these to be inlined means you lose the reference linking which happens in spec now which is useful to track down usages of things.

Reply:
@dj2 Usage is only once here though, clicking that 'type' mostly creates a loss of energy thinking it will take you to a place with more explanation about it.

Reply:
Well, it tells you what the keyword resolves too. But the more useful is from the keyword section click on type brings you back to the usage in type alias. So, you can see where it's used.